protocol/parser/parser.go

```diff
@@ -122,105 +123,120 @@ func parseInputFromParamsWithGenericParsers(rpcInput RPCInput, genericParsers []
+	resString, ok := result[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)
 	if !ok {
-		return "", utils.LavaFormatError("Failed to Convert blockData[spectypes.DEFAULT_PARSED_RESULT_INDEX].(string)", nil, utils.Attribute{Key: "blockData", Value: response[spectypes.DEFAULT_PARSED_RESULT_INDEX]})
+		return "", utils.LavaFormatDebug("blockParsing - result[0].(string) - type assertion failed", utils.LogAttr("result[0]", result[0]))
 	}
 
-	if strings.Contains(response, "\"") {
-		responseUnquoted, err := strconv.Unquote(response)
-		if err != nil {
-			return response, nil
-		}
-		return responseUnquoted, nil
+	return resString, nil
+}
+
+func parseBlock(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser, source int) *ParsedInput {
+	parsedBlockInfo, parsedSuccessfully := parseInputWithGenericParsers(rpcInput, genericParsers)
+	if parsedSuccessfully {
+		parsedBlockInfo.parsedDataRaw = unquoteString(parsedBlockInfo.parsedDataRaw)
+		return parsedBlockInfo
+	}
+	if parsedBlockInfo == nil {
+		parsedBlockInfo = NewParsedInput()
 	}
 
-	return response, nil
+	parsedRawBlock, _ := parseInputWithLegacyBlockParser(rpcInput, blockParser, source)
+	parsedBlockInfo.parsedDataRaw = unquoteString(parsedRawBlock)
+	return parsedBlockInfo
+}
+
+func ParseRawBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) *ParsedInput {
+	return parseBlock(rpcInput, blockParser, genericParsers, PARSE_PARAMS)
 }
 
-func ParseBlockFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser) (int64, error) {
-	result, err := ParseFromReply(rpcInput, blockParser)
+func ParseBlockFromParams(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) *ParsedInput {
+	parsedInput := parseBlock(rpcInput, blockParser, genericParsers, PARSE_PARAMS)
+	ParseRawBlock(rpcInput, parsedInput, blockParser.DefaultValue)
+	return parsedInput
+}
+
+func ParseBlockFromReply(rpcInput RPCInput, blockParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) *ParsedInput {
+	parsedInput := parseBlock(rpcInput, blockParser, genericParsers, PARSE_RESULT)
+	ParseRawBlock(rpcInput, parsedInput, blockParser.DefaultValue)
+	return parsedInput
+}
+
+func unquoteString(str string) string {
+	if !strings.Contains(str, "\"") {
+		return str
+	}
+
+	unquoted, err := strconv.Unquote(str)
 	if err != nil {
-		return spectypes.NOT_APPLICABLE, err
+		return str
 	}
-	return rpcInput.ParseBlock(result)
+	return unquoted
 }
 
 // This returns the parsed response after decoding
-func ParseFromReplyAndDecode(rpcInput RPCInput, resultParser spectypes.BlockParser) (string, error) {
-	response, err := ParseFromReply(rpcInput, resultParser)
+func ParseBlockHashFromReplyAndDecode(rpcInput RPCInput, resultParser spectypes.BlockParser, genericParsers []spectypes.GenericParser) (string, error) {
+	parsedInput, _ := parseInputWithGenericParsers(rpcInput, genericParsers)
```

Comment:
why on hashes we don't make sure the parsedSuccessfuly is set and on regular parsing it does, thats a discrepancy in how it works

Reply:
Same as previous comment
> It doesn't matter if we parsed hashes or not, parsedSuccessfully will be true only if we parsed a raw block successfully.